src/NuGet.Core/NuGet.Protocol/Plugins/PluginDiscoverer.cs

```diff
@@ -140,6 +143,105 @@ private List<PluginFile> GetPluginFiles(CancellationToken cancellationToken)
             return files;
         }
 
+        /// <summary>
+        /// Gets auth plugins installed using dotnet tools. This is done by iterating through each file in directories found int eh
+        /// `PATH` environment variable.
+        /// The files are also expected to have a name that starts with `nuget-plugin-`
+        /// </summary>
+        /// <returns></returns>
+        private static List<string> GetNetToolsPluginFiles()
+        {
+            var pluginFiles = new List<string>();
+            var paths = Environment.GetEnvironmentVariable("PATH")?.Split(Path.PathSeparator) ?? Array.Empty<string>();
+
+            foreach (var path in paths)
+            {
+                if (Directory.Exists(path))
+                {
+                    var directoryInfo = new DirectoryInfo(path);
+                    var files = directoryInfo.GetFiles("nuget-plugin-*");
+
+                    foreach (var file in files)
+                    {
+                        if (IsValidPluginFile(file))
+                        {
+                            pluginFiles.Add(file.FullName);
+                        }
+                    }
+                }
+            }
+
+            return pluginFiles;
+        }
+
+        /// <summary>
+        /// Checks whether a file is a valid plugin file for windows/Unix.
+        /// Windows: It should be either .bat or  .exe
+        /// Unix: It should be executable
+        /// </summary>
+        /// <param name="fileInfo"></param>
+        /// <returns></returns>
+        private static bool IsValidPluginFile(FileInfo fileInfo)
+        {
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return fileInfo.Extension.Equals(".exe", StringComparison.OrdinalIgnoreCase) ||
+                       fileInfo.Extension.Equals(".bat", StringComparison.OrdinalIgnoreCase);
+            }
+            else
+            {
+#if NET8_0
```

Comment:
```suggestion
#if NET8_0_OR_HIGHER
```

Otherwise when the project is retarged to a higher TFM in the future, this block will no longer be used